app/src/main/java/com/epfl/drawyourpath/notifications/AlarmReceiver.kt

```diff
@@ -0,0 +1,74 @@
+package com.epfl.drawyourpath.notifications
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.login.ENABLE_ONETAP_SIGNIN
+import com.epfl.drawyourpath.login.LoginActivity
+import com.epfl.drawyourpath.login.RESTORE_USER_IN_KEYCHAIN
+import com.epfl.drawyourpath.mainpage.USE_MOCK_CHALLENGE_REMINDER
+
+/**
+ * Class that can execute code when it receives an alarm (with onReceive())
+ */
+class AlarmReceiver : BroadcastReceiver() {
+
+    /**
+     * Sends a notification when an alarm is received
+     */
+    override fun onReceive(context: Context, intent: Intent) {
+        val notificationManager = ContextCompat.getSystemService(
+            context,
+            NotificationManager::class.java
+        ) as NotificationManager
+
+        val disableAutoReconnection = intent.getBooleanExtra(USE_MOCK_CHALLENGE_REMINDER, false)
+
+        notificationManager.sendChallengeReminderNotification(context, !disableAutoReconnection)
+        // Send other time-based notifications here
+    }
```

Comment:
Question: Does this function know which notification to send when the alarm is received or will it launch ervery notifications every time an alarm is received? 
(in the case where we add new notifications and add new reminders)